backend/api/views.py

```diff
@@ -0,0 +1,399 @@
+
+                return (text_en, text_fr, text_es)
+
+            except sr.UnknownValueError:
+                details = {"User_Request": " ",
+                           "Chatbot_Response": "No audio input given. Please press the button to speak to BabelBot!"}
+                return render(request,"kiosk.html", details)
+            
+            except sr.RequestError as e:
+                details = {"User_Request": " ",
+                           "Chatbot_Response": f"Could not request results from Google Speech Recognition service; {e}"}
+                return render(request,"kiosk.html", details)
+            
+    return HttpResponse("Method Not Allowed")
+
+
+def lang_detect(response1, response2, response3):
+    """
+    Function for detecting the language and its probability
+    ie. [["es:0.9999932356592219"], ["es:0.9999941641326395"], ["es:0.999996766257174"], ["ne:0.5714284946155643", "hi:0.4285697429223616"]]
+    """
+    language_prob = []
+
+    # Make into string elements for data cleaning in all 3 languages
+    detect_en = [str(i) for i in detect_langs(response1)]
+    language_prob.extend(detect_en)
+
+    detect_fr = [str(i) for i in detect_langs(response2)]
+    language_prob.extend(detect_fr)
+
+    detect_es = [str(i) for i in detect_langs(response3)]
+    language_prob.extend(detect_es)
+
+    return list(language_prob)
+
+
+def lang_prob(lang_probability):
+    """
+    Function for seperating the probability and language codes
+    """
+    language_code = []
+
+    # Seperate probs and language codes into 2 lists
+    for i in range(0, len(lang_probability)):
+        code = lang_probability[i][0:2]
+        language_code.append(code)
+        lang_probability[i] = lang_probability[i][3:]
+
+    # Flatten list
+    probs = [float(i) for i in lang_probability]
+
+    return language_code, probs
+
+
+def ISO_639(langauge_code, probability):
+    """
+    Function to get the language ISO
+    """
+    lang_counter = Counter(langauge_code)
+
+    test_val = list(lang_counter.values())[0]
+    res = True
+    
+    for i in lang_counter:
+        if lang_counter[i] != test_val:
+            res = False
+            break
+
+    if res == True:
+        d = defaultdict(list)
+        for key, value in zip(langauge_code, probability):
+            d[key].append(value)
+
+        mean_probs = {key: np.mean(val) for key, val in dict(d).items()}
+
+        ISO_639_2 = str(max(mean_probs, key = mean_probs.get))
+
+    else:
+        ISO_639_2 = max(lang_counter, key = lang_counter.get)
+    
+    return ISO_639_2
+
+
+def exit_input(message):
+    if message.lower() == "exit please":
+        return 1
+    return 0
+
+
+def speak_response(lang_ISO, res_en2lang, lang_voice):
+    """
+    Translate audio response into different languages
+    """
+    if lang_ISO in lang_voice:
+        engine = pyttsx3.init()
+        if lang_ISO == "es":
+            engine.setProperty("rate", 150)  # Speed of speech (words per minute)
```

Comment:
Should we set this for other languages too? Or is this language the only one that sounded to fast/slow by default

Reply:
Yes, so Spanish was the only language (from English and French) that sounded too fast, hence, I had to change the speed of speech.